feat: tabs component

andreancardona opened the pull request:
Closes #384 

- this PR adds the tabs to asset detail page. 

## Sketch file redlines: 

- [Box file: Component Asset Detail Page Redlines](https://ibm.ent.box.com/folder/150822549524)

## Note:

- The tabs for v11 do not receive hover state at the moment (known. issue) current PR does not contain hover styles

andreancardona committed:
feat: added tabs with placeholder contnent

andreancardona committed:
feat: override carbon style updates

alisonjoseph commented:
This is a design question, why are we not using Carbon tabs out of the box?

Quick local test of what it looks like with default Carbon styles
<img width="531" alt="Screen Shot 2022-03-16 at 9 50 01 PM" src="https://user-images.githubusercontent.com/2753488/158726688-1f7dcb28-68de-4945-ae62-ee2647f3d4df.png">

alisonjoseph commented on services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].module.scss:
```diff
@@ -35,3 +35,37 @@
+  }
+}
+
+//override Carbon min height and border bottom styles
+.tab-list {
+  min-height: $spacing-09;
+
+  :global(.cds--tabs__nav-link) {
+    max-width: fit-content;
+
+    border-bottom: $spacing-02 solid var(--tab-border-color, transparent);
+    color: $text-primary;
+    @include type-style('body-short-02', true);
+  }
+
+  :global(.cds--tabs__nav-item--selected) {
+    border-bottom: $spacing-02 solid $border-interactive;
+    font-weight: 600;
+  }
+}
```
I'd recommend removing all of this and using core Carbon styles. (need to sync w/ design)

alisonjoseph commented on services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].module.scss:
```diff
@@ -35,3 +35,37 @@
+    z-index: 0;
+    left: 0;
+    width: 100vw;
+    height: $spacing-09;
+    border-bottom: 1px solid $layer-active-01;
```
Would we want to just use the pixel height here?. For the border-bottom, is there a border token that would make more sense?

francinelucca reviewed (dismissed):
Looks visually correct to me.
<img width="1395" alt="Content block horizontal" src="https://user-images.githubusercontent.com/40550942/158842209-faf634bc-8c64-4f38-b101-e899b9602e53.png">

Just comment on repeated tabs, if design says we're good with carbon overrides then LGTM :tada

mattrosno reviewed (commented):
- [ ] This right chevron isn't clickable?

<img width="372" alt="image" src="https://user-images.githubusercontent.com/1691245/158846641-b31a704b-fea3-4002-8eca-725a8a901fcb.png">

- [x] Should the tab bar be sticky on scroll (immediately under global nav)?
- [x] Are there supposed to be hover states?
- [x] Can we put these in a new `PageTabs` component that passes in an array of tabs?

Something like this makes sense to me:

```jsx
PageTabs.propTypes = {
  className: PropTypes.string,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      path: PropTypes.string.isRequired
    })
  )
}
```

mattrosno reviewed (commented):
Should page tabs stack like this on scroll? @kingtraceyj 

<img width="967" alt="image" src="https://user-images.githubusercontent.com/1691245/158849129-9d76a38c-13a6-4beb-bab2-89a03c8b22ab.png">

kingtraceyj commented:
yes, those tabs are sticky on scroll like they are today on the site.

andreancardona commented on services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js:
```diff
@@ -87,6 +87,20 @@ const Asset = ({ libraryData }) => {
+              <Tab>Usage</Tab>
+              <Tab>Design</Tab>
+              <Tab>Code</Tab>
+              <Tab>Usage</Tab>
+              <Tab>Design</Tab>
+              <Tab>Code</Tab>
```
si. :)

andreancardona commented:
> * an we put these in

- I have a note in the PR description but Tabs in v11 have a hover bug (known). Will inquire more. :) 
- Not sure what's going on with the click scroll in mobile - will also inquire

- [x] Move tabs to `PageTabs` component - sorry about this should have done this from the start

andreancardona committed:
feat: added tabs to PageTabs

andreancardona committed:
feat: need to fix hover styles

alisonjoseph commented on services/web-app/components/page-tabs/page-tabs.module.scss:
```diff
@@ -0,0 +1,45 @@
+  }
+}
+
+// override Carbon min height and border bottom styles
+.tab-list {
+  min-height: $spacing-09;
+
+  // TODO: fix hover / active styles
+  :global(.cds--tabs__nav-item--selected) {
+    border-bottom: $spacing-02 solid $border-interactive;
+    font-weight: 600;
+  }
+
+  :global(.cds--tabs__nav-link) {
+    max-width: fit-content;
+
+    border-bottom: $spacing-02 solid var(--tab-border-color, transparent);
+    color: $text-secondary;
+    @include type-style('body-short-02', true);
+
+    &:hover {
+      border-bottom: $spacing-02 solid $layer-active-01;
+      color: $text-primary;
+    }
+  }
+}
```
```suggestion
:global(.cds--tabs .cds--tab--list) {
  width: auto;
}
```

alisonjoseph commented on services/web-app/components/page-tabs/page-tabs.module.scss:
```diff
@@ -0,0 +1,45 @@
+    z-index: 0;
+    left: 0;
+    width: 100vw;
+    height: $spacing-09;
```
```suggestion
    height: $spacing-08;
```

to match default tab height

andreancardona committed:
feat: updated styles with hover

andreancardona committed:
Merge branch 'main' into 384-tabs-component

andreancardona committed:
Update services/web-app/components/page-tabs/page-tabs.js

Co-authored-by: Alison Joseph <alisonejoseph@Gmail.com>

andreancardona committed:
feat: update style imports

andreancardona committed:
Merge branch '384-tabs-component' of https://github.com/carbon-design-system/carbon-platform into 384-tabs-component

andreancardona commented on services/web-app/components/page-tabs/page-tabs.module.scss:
```diff
@@ -0,0 +1,45 @@
+//
+.container {
+  @include breakpoint(md) {
+    margin-left: -$spacing-05;
```
yess thank you, changes made!

andreancardona commented:
@kingtraceyj 

This is where the tabs preview is at now. There are 2 known bugs with Carbon v11 Tabs. 
1. Text is _not_ left-aligned (currently centered)

2. On focus text jumps up by 1px because padding is not embedded. 

Let me know if there is anything else you see that needs to be updated

https://user-images.githubusercontent.com/32720851/159508353-be1e252c-188d-4c3a-8fe0-c51e7a993379.mov

andreancardona committed:
feat: fixed link large styling and updated styles

alisonjoseph commented on services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js:
```diff
@@ -111,11 +115,13 @@ const Asset = ({ libraryData }) => {
-                    {`v${libraryData.content.version}`}
-                  </a>
-                </Link>
+                <CarbonLink
```
Tabs look great! I think we need to wrap the CarbonLink in the Next js Link component, is that correct @mattrosno 

I actually just opened a small PR here to fix the styles https://github.com/carbon-design-system/carbon-platform/pull/407

andreancardona replied:
and so in this case - yep they would need to be wrapped. I'll remove my update from this PR so it doesn't conflict with yours, which is the correct solution.  :)

andreancardona committed:
fix: reverting link style changes

to avoid merge conflicts this is being fixed here: https://github.com/carbon-design-system/carbon-platform/pull/407

alisonjoseph reviewed (dismissed):
🥳 LGTM!

andreancardona commented:
Things to note: 

- Currently the scroll click in mobile does not work - the bug is known and the issue is reported: https://github.com/carbon-design-system/carbon/issues/11037

andreancardona committed:
fix: fixed left margin

andreancardona committed:
Merge branch '384-tabs-component' of https://github.com/carbon-design-system/carbon-platform into 384-tabs-component

francinelucca commented on services/web-app/components/page-tabs/page-tabs.module.scss:
```diff
@@ -0,0 +1,53 @@
+  }
+
+  /* stylelint-disable-next-line no-descending-specificity */
+  :global(.cds--tabs__nav-item--selected) {
```
is this a carbon override? should we add an override comment like in https://github.com/carbon-design-system/carbon-platform/blob/main/services/web-app/components/catalog-sort/catalog-sort.module.scss#L79

francinelucca reviewed (dismissed):
🎉

mattrosno commented:
@andreancardona can you request new reviews after you merge in main branch?

andreancardona commented:
> I'm seeing this locally now after pulling in main.
> 
> ![Screen Shot 2022-03-24 at 12 28 37 PM](https://user-images.githubusercontent.com/2753488/159975465-b8d2ceb3-bc90-4e17-a828-43dc7a708f40.png)

hmm looking into this now

andreancardona committed:
Merge branch 'main' into 384-tabs-component

andreancardona committed:
Merge branch 'main' into 384-tabs-component

andreancardona committed:
Merge remote-tracking branch 'origin/main' into 384-tabs-component

andreancardona committed:
Merge branch '384-tabs-component' of https://github.com/carbon-design-system/carbon-platform into 384-tabs-component

andreancardona committed:
feat: fix page tab import

andreancardona commented:
@alisonjoseph fixed - ready for your review!

alisonjoseph reviewed (dismissed):
LGTM!

andreancardona committed:
Merge branch 'main' into 384-tabs-component

jdharvey-ibm commented on services/web-app/components/page-tabs/page-tabs.module.scss:
```diff
@@ -0,0 +1,49 @@
+//
+// Copyright IBM Corp. 2022, 2022
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
```
```suggestion
/*
 * Copyright IBM Corp. 2022, 2022
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
```

jdharvey-ibm commented on services/web-app/components/page-tabs/page-tabs.module.scss:
```diff
@@ -0,0 +1,49 @@
+    }
+  }
+
+  /* stylelint-disable-next-line no-descending-specificity */
+  :global(.cds--tabs__nav-item--selected) {
+    border-bottom: spacing.$spacing-02 solid theme.$border-interactive;
+    @include type.type-style('heading-02');
+
+    color: theme.$text-primary;
+  }
```
Could you add a comment for why this override is needed? I read through the stylelint docs for it and it looks like maybe this is a limitation of stylelint? It seems to not be recognizing that the two global blocks are actually targeting different class names.

andreancardona committed:
Update services/web-app/components/page-tabs/page-tabs.module.scss

Co-authored-by: Joe Harvey <51208233+jdharvey-ibm@users.noreply.github.com>

andreancardona committed:
update stylint override
